index.bs

```diff
@@ -10093,55 +10088,76 @@ Dictionary {{AudioWorkletNodeOptions}} Members</h6>
 
-With a combination of {{AudioWorkletNodeOptions/numberOfInputs}},
-{{AudioWorkletNodeOptions/numberOfOutputs}},
-and {{AudioWorkletNodeOptions/outputChannelCount}},
-various channel configurations can be achieved.
-
-: {{AudioWorkletNodeOptions/numberOfInputs}} = 0,
-	{{AudioWorkletNodeOptions/numberOfOutputs}} = 0
-:: {{NotSupportedError}} MUST be thrown by the constructor.
-
-: {{AudioWorkletNodeOptions/numberOfInputs}} = 1,
-	{{AudioWorkletNodeOptions/numberOfOutputs}} = 1
-:: If {{AudioWorkletNodeOptions/outputChannelCount}} is
-	unspecified, the output channel count will match
-	<a>computedNumberOfChannels</a> from the input. If
-	{{AudioWorkletNodeOptions/outputChannelCount}} is specified
-	with one value, the output channel count will match the given value.
-
-: All other cases
-:: If {{AudioWorkletNodeOptions/outputChannelCount}} is
-	unspecified, it will be mono for all outputs. If specified,
-	the values will be used as given.
+The following algorithm describes how an {{AudioWorkletNodeOptions}} can be
+used to configure various channel configurations.
+
+<div id="configure-with-audioworkletnodeoptions" algorithm="configure with AudioWorkletNodeOptions">
+	1. Let <var>node</var> be an {{AudioWorkletNode}} instance that is
+		given to this algorithm.
+
+	1. If both {{AudioWorkletNodeOptions/numberOfInputs}} and
+		{{AudioWorkletNodeOptions/numberOfOutputs}} are zero,
+		throw a {{NotSupportedError}} and abort the remaining steps.
+
+	1. If {{AudioWorkletNodeOptions/outputChannelCount}} is
+		<a spec="webidl" lt="present">present</a>,
+
+		1. If any value in
+			{{AudioWorkletNodeOptions/outputChannelCount}} is zero
+			or greater than the implementation’s maximum number
+			of channels, throw a {{NotSupportedError}} and abort
+			the remaining steps.
+
+		1. If the length of
+			{{AudioWorkletNodeOptions/outputChannelCount}} does not
+			match {{AudioWorkletNodeOptions/numberOfOutputs}},
+			throw an {{IndexSizeError}} and abort the remaining
+			steps.
+
+		1. If both {{AudioWorkletNodeOptions/numberOfInputs}} and
+			{{AudioWorkletNodeOptions/numberOfOutputs}} are 1,
+			set the channel count of the <var>node</var> output to
+			the one value in
+			{{AudioWorkletNodeOptions/outputChannelCount}}.
+
+		1. Otherwise set the channel count of each output of the
+			<var>node</var> to the corresponding value in the
+			{{AudioWorkletNodeOptions/outputChannelCount}} sequence
+			and return.
+
+	1. If {{AudioWorkletNodeOptions/outputChannelCount}} is
+		<a spec="webidl" lt="present">not present</a>,
+
+		1. If both {{AudioWorkletNodeOptions/numberOfInputs}} and
+			{{AudioWorkletNodeOptions/numberOfOutputs}} are 1,
+			set the channel count of the <var>node</var> output to
+			1 and return.
+
+			NOTE: For this case, the output chanel count will
+			change to <a>computedNumberOfChannels</a> dynamically
+			based on the input at runtime.
+
+		1. Otherwise set the channel counts of the outputs of the
```

Comment:
Perhaps better phrased as "set the channel count of each output of the node to 1..."